### ISSUE TITLE
Bun.serve ERR_INVALID_ARG_TYPE on Linux x64 baseline

### ISSUE THREAD

**Issue by jaromanda** [oven-sh/bun#12703]
### What version of Bun is running?

v1.1.20

### What platform is your computer?

Linux 6.9.7+bpo-amd64 x86_64 unknown

### What steps can reproduce the bug?

Try the simple http server code from https://bun.sh/guides/http/simple

### What is the expected behavior?

The server runs

### What do you see instead?

```
1 | const server = Bun.serve({
                       ^
TypeError: 湕捩摯⁥潨瑳慮敭⁳畭瑳愠牬慥祤戠⁥湥潣敤⁤潦⁲潮⹷渊睥唠䱒椨灮瑵⸩潨瑳慮敭猠潨汵⁤潤琠敨琠楲正.敒摡扡敬瑓敲浡椠⁳湩椠癮污摩猠慴整.湉整湲污挠湯楳瑳湥祣攠牲牯›慷捴敨⁲畭整⁸獩氠捯敫⁤桷湥椠
 code: "ERR_INVALID_ARG_TYPE"

Bun v1.1.20 (Linux x64 baseline)
```

### Additional information

The cpu has no AVX2 instruction, have installed bun Linux x64 baseline

Code works fine on "modern" cpu, but that is not an option for this VPS server.

Not sure if it's relevant, but if I cross compile i.e. using `--compile=bun-linux-x64-baseline` - the generated executable runs on `modern` cpu, but results in

    error: ENOENT reading ""

On the baseline system - regardless of where I `compile` from (modern or baseline)

Edit: I've noticed that the above error isn't consistent

    error: ENOENT reading "`x|root/baseline-app"
    error: ENOENT reading "`t!root/baseline-app"
    error: ENOENT reading "`Uroot/baseline-app"

Are examples of other outputs

**Comment by YANGLEDUO1** [comment]
Are you running in a virtual machine?

**Comment by jaromanda** [comment]
Yes, but as I said
> "Code works fine on "modern" cpu, but that is not an option for this **VPS** server.

**Comment by YANGLEDUO1** [comment]
The cpu type should not be recognized. I am trying to modify the correct cpu, because I also have this problem

**Comment by YANGLEDUO1** [comment]
It's my guess.

**Comment by jaromanda** [comment]
This seems to be fixed with v1.1.22 - thank you
